datasets/flwr_datasets/partitioner/pathological_partitioner.py

```diff
@@ -0,0 +1,295 @@
+                f"of unique classes in the given dataset={num_unique_classes}. "
+                f"Reduce the `num_classes_per_partition` or make use different dataset "
+                f"to apply this partitioning."
+            )
+        if self._class_assignment_mode == "first-deterministic":
+            # if self._first_class_deterministic_assignment:
+            for partition_id in range(self._num_partitions):
+                label = partition_id % num_unique_classes
+                self._partition_id_to_unique_labels[partition_id].append(label)
+
+                while (
+                    len(self._partition_id_to_unique_labels[partition_id])
+                    < self._num_classes_per_partition
+                ):
+                    label = self._rng.choice(self._unique_labels, size=1)[0]
+                    if label not in self._partition_id_to_unique_labels[partition_id]:
+                        self._partition_id_to_unique_labels[partition_id].append(label)
+        elif self._class_assignment_mode == "deterministic":
+            for partition_id in range(self._num_partitions):
+                labels = []
+                for i in range(self._num_classes_per_partition):
+                    label = self._unique_labels[
+                        (partition_id + i) % len(self._unique_labels)
+                    ]
+                    labels.append(label)
+                self._partition_id_to_unique_labels[partition_id] = labels
+        elif self._class_assignment_mode == "random":
+            for partition_id in range(self._num_partitions):
+                labels = self._rng.choice(
+                    self._unique_labels,
+                    size=self._num_classes_per_partition,
+                    replace=False,
+                ).tolist()
+                self._partition_id_to_unique_labels[partition_id] = labels
+        else:
+            raise ValueError(
+                f"The supported class_assignment_mode are: 'random', 'deterministic', "
+                f"'first-deterministic'. You provided: {self._class_assignment_mode}."
+            )
+
+    def _determine_unique_label_to_times_used(self) -> None:
+        """Determine how many times the label is used in the partitions.
+
+        This computation is based on the assigment of the label to the partition_id in
+        the `_determine_partition_id_to_unique_labels` method.
+        """
+        for unique_label in self._unique_labels:
+            self._unique_label_to_times_used_counter[unique_label] = 0
+        for unique_labels in self._partition_id_to_unique_labels.values():
+            for unique_label in unique_labels:
+                self._unique_label_to_times_used_counter[unique_label] += 1
+
+    def _check_correctness_of_unique_label_to_times_used_counter(
+        self, labels: NDArray
+    ) -> None:
+        """Check if the number of times the label is possible to execute.
```

Comment:
```suggestion
        """Check if the number of times the label assigned to a partition.
```

Reply:
i think it's a bit clearer like this, but I'll leave it up to you. Probably it can be tweaked further

Reply:
```suggestion
        "Check if partitioning is possible given the presence requirements.
```

Reply:
I can propose alternatively this.

Reply:
looks good!